metis-enrichment/metis-enrichment-client/src/test/java/eu/europeana/enrichment/rest/client/EnrichmentWorkerTest.java

```diff
@@ -0,0 +1,236 @@
+    final List<EnrichmentResultList> expectedMerges = new ArrayList<>();
+    if (doDereferencing) {
+      expectedMerges
+          .addAll(dereferenceResult.stream().filter(Objects::nonNull).collect(Collectors.toList()));
+    }
+    if (doEnrichment) {
+      expectedMerges.add(enrichmentResult);
+    }
+    verify(entityMergeEngine, times(expectedMerges.size())).mergeEntities(any(),
+        enrichmentResultCaptor.capture());
+    // Note that the captor returns a linked list, so we don't want to use indices.
+    // But the interface gives a generic type List, so we don't want to depend on the
+    // linked list functionality either.
+    int currentPointer = 0;
+    for (List<EnrichmentBase> capturedMerge : enrichmentResultCaptor.getAllValues()) {
+      assertArrayEquals(expectedMerges.get(currentPointer).getResult().toArray(),
+          capturedMerge.toArray());
+      currentPointer++;
+    }
+  }
+
+  @Test
+  public void testProcessWrapperMethods()
+      throws JiBXException, UnsupportedEncodingException, DereferenceOrEnrichException {
+
+    // Create enrichment worker and mock the actual worker method as well as the RDF conversion
+    // methods.
+    final EnrichmentWorker worker = spy(new EnrichmentWorker(null, null, null));
+    final RDF inputRdf = new RDF();
+    final String outputString = "OutputString";
+    doReturn(inputRdf).when(worker).convertStringToRdf(anyString());
+    doReturn(outputString).when(worker).convertRdfToString(inputRdf);
+    doReturn(inputRdf).when(worker).process(any(), any());
+
+    // Perform the operations and verify the result
+    final RDF returnedRdf = worker.process(inputRdf);
+    assertEquals(inputRdf, returnedRdf);
+    final String returnedString = worker.process("");
+    assertEquals(outputString, returnedString);
+
+    // Validate the method calls to the actual worker method
+    verify(worker, times(2)).process(any(), any());
+    verify(worker, times(2)).process(inputRdf, Mode.DEREFERENCE_AND_ENRICHMENT);
+
+    // Test null string input
+    try {
+      worker.process((String) null);
+      fail("Expected an exception to occur.");
+    } catch (IllegalArgumentException e) {
+      // This is expected
+    }
+  }
+
+  @Test
+  public void testEnrichmentWorkerNullValues() throws DereferenceOrEnrichException {
+
+    // Create enrichment worker
+    final EnrichmentWorker worker = new EnrichmentWorker(null, null, null);
+
+    // Test null string input
+    try {
+      worker.process(null, Mode.DEREFERENCE_AND_ENRICHMENT);
+      fail("Expected an exception to occur.");
+    } catch (IllegalArgumentException e) {
+      // This is expected
+    }
+
+    // Test null string input
+    try {
+      worker.process(new RDF(), null);
+      fail("Expected an exception to occur.");
+    } catch (IllegalArgumentException e) {
+      // This is expected
+    }
+  }
+
+  private static <T> ArgumentCaptor<List<T>> createCaptorForList() {
```

Comment:
This seems an overkill method. Do inline initialization or maybe use something like the @Captor annotation on a field